integreat_cms/static/src/js/forms/update-permalink.ts

```diff
@@ -86,4 +86,19 @@ window.addEventListener("load", () => {
         slugField.value = currentSlug;
         toggleSlugMode();
     });
+
+    // This is used to slugify the title of the slug in target language in the page_sbs template
+    document.getElementById("target_translation_title")?.addEventListener("focusout", () => {
+        const submissionLock = new SubmissionPrevention(".no-premature-submission");
+        const targetTitleElement = (document.getElementById("target_translation_title") as HTMLInputElement).value;
+        const targetLinkElement = document.getElementById("target_slug-link") as HTMLInputElement;
+        const dataset = (document.querySelector('[for="id_slug"]') as HTMLElement).dataset;
+        slugify(dataset.slugifyUrl, { title: targetTitleElement, model_id: dataset.modelId })
+            .then((response) => {
+                /* on success write response to both slug field and permalink */
+                targetLinkElement.value = response.unique_slug;
+                updatePermalink(response.unique_slug);
+            })
+            .finally(() => submissionLock.release());
+    });
```

Comment:
This is (almost) exact same function from line 47ff:
```typescript
        document.getElementById("id_title").addEventListener("focusout", ({ target }) => {
            const submissionLock = new SubmissionPrevention(".no-premature-submission");
            const currentTitle = (target as HTMLInputElement).value;
            const dataset = (document.querySelector('[for="id_title"]') as HTMLElement).dataset;
            slugify(dataset.slugifyUrl, { title: currentTitle, model_id: dataset.modelId })
                .then((response) => {
                    /* on success write response to both slug field and permalink */
                    slugField.value = response.unique_slug;
                    updatePermalink(response.unique_slug);
                })
                .finally(() => submissionLock.release());
        });
    }
```

It's generally not super maintainable to have such duplicate pieces of code. Do you think you could write a function that accomodates both usecases, i.e. is called with arguments for the element ID and the target element?

Reply:
@charludo you are completely right, I have now merged them into one :smiley_cat: